pkg/executor/util.go

```diff
@@ -142,3 +144,28 @@ func createPortNameForService(
 	}
 	return strings.ToLower(portInfo.URLScheme) + "-" + portInfo.ID
 }
+
+// ValidateName is a utility function that limits object names to be below
+// nameLengthLimit threshold for the CrNameRole naming scheme.
+func ValidateName(
+	name string,
+) string {
+	length := len(name)
+	var modName string
+
+	if length == 0 {
+		return name
+	}
+
+	for i := 0; i < length && i < nameLengthLimit; i++ {
```

Comment:
If the limit is calculated as in my other comment then this loop could do "<= nameLengthLimit" instead of "< nameLengthLimit".